File: src/main/java/solve/GameState.java
package solve;

public class GameState<P> {

    private int remoteness = -1;
    private int state = -1;
    private final P pos;

    public GameState(P pos) {
        this.pos = pos;
    }

    public GameState(P pos, int remoteness, int state) {
        this.pos = pos;
        this.remoteness = remoteness;
        this.state = state;
    }

}

File: src/main/kotlin/GameState.kt
package solve;

class GameState<P>(pos: P) {

    var remoteness = -1;
    var state = -1;
    var pos: P = pos;

    constructor(pos: P, remoteness: Int, state: Int) : this(pos) {
        this.remoteness = remoteness;
        this.state = state;
    }

}
